src/std/actor-v18/proto.ss

```diff
@@ -0,0 +1,97 @@
+;;; -*- Gerbil -*-
+;;; © vyzo
+;;; actor protocol messages
+(import :gerbil/gambit/threads
+        :std/sugar
+        ./message)
+(export #t)
+
+(defmessage !ok (value))
+(defmessage !error (message))
+
+(defrules with-result ()
+  ((_ expr fail!)
+   (match expr
+     ((!ok value) value)
+     ((!error what) (fail! what))))
+  ((_ expr)
+   (with-result expr error)))
+
+(defrules defcall-actor ()
+  ((_ (proc arg ...) expr error: error-msg error-irritant ...)
+   (def (proc arg ...)
+     (with-result expr
+       (lambda (what) (raise-actor-error 'proc error-msg error-irritant ... what)))))
```

Comment:
Is the "what" given by the remote actor? That deserves to be included in the documentation... also is a potential issue if logged without proper escaping (or resulting in overflow and partial logging)

Reply:
it is a string mostly, but it can really be anything; so nothing to document here.

Reply:
for the issue: no, I don't think so.